e2e/e2etests/test_message_passing_zevm.go

```diff
@@ -0,0 +1,332 @@
+	tx, err = testDAppZEVM.SendHelloWorld(r.ZEVMAuth, destinationAddress, EVMChainID, amount, false)
+	if err != nil {
+		panic(err)
+	}
+	r.Logger.Info("TestDApp.SendHello tx hash: %s", tx.Hash().Hex())
+	receipt = utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+	if receipt.Status == 0 {
+		panic(fmt.Sprintf("send failed, logs: %+v", receipt.Logs))
+	}
+
+	// Transaction is picked up by the zetanode evm hooks and a new contract call is initiated on the EVM chain
+	cctx := utils.WaitCctxMinedByInTxHash(r.Ctx, receipt.TxHash.String(), r.CctxClient, r.Logger, r.CctxTimeout)
+	if cctx.CctxStatus.Status != cctxtypes.CctxStatus_OutboundMined {
+		panic("expected cctx to be outbound_mined")
+	}
+	r.Logger.Print(fmt.Sprintf("🔄 Cctx mined for contract call chain zevm %s", cctx.Index))
+
+	// On finalization the Tss calls the onReceive function which in turn calls the onZetaMessage function on the destination contract.
+	receipt, err = r.EVMClient.TransactionReceipt(r.Ctx, ethcommon.HexToHash(cctx.GetCurrentOutTxParam().OutboundTxHash))
+	if err != nil {
+		panic(err)
+	}
+	if receipt.Status != 1 {
+		panic("tx failed")
+	}
+	testDAppEVM, err := testdapp.NewTestDApp(r.EvmTestDAppAddr, r.EVMClient)
+	if err != nil {
+		panic(err)
+	}
+	receivedHelloWorldEvent := false
+	for _, log := range receipt.Logs {
+		_, err := testDAppEVM.ParseHelloWorldEvent(*log)
+		if err == nil {
+			r.Logger.Info("Received HelloWorld event:")
+			receivedHelloWorldEvent = true
+		}
+	}
+	if !receivedHelloWorldEvent {
+		panic("expected HelloWorld event")
+	}
+}
+
+func TestMessagePassingZEVMtoEVMRevert(r *runner.E2ERunner, args []string) {
+	if len(args) != 1 {
+		panic("TestMessagePassing requires exactly one argument for the amount.")
+	}
+
+	amount, ok := big.NewInt(0).SetString(args[0], 10)
+	if !ok {
+		panic("Invalid amount specified for TestMessagePassing.")
+	}
+
+	// Set destination details
+	EVMChainID, err := r.EVMClient.ChainID(r.Ctx)
+	if err != nil {
+		panic(err)
+	}
+	destinationAddress := r.EvmTestDAppAddr
+
+	// Contract call originates from ZEVM chain
+	r.ZEVMAuth.Value = amount
+	tx, err := r.WZeta.Deposit(r.ZEVMAuth)
+	if err != nil {
+		panic(err)
+	}
+	r.ZEVMAuth.Value = big.NewInt(0)
+	r.Logger.Info("wzeta deposit tx hash: %s", tx.Hash().Hex())
+	receipt := utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+	r.Logger.EVMReceipt(*receipt, "wzeta deposit")
+	if receipt.Status == 0 {
+		panic("deposit failed")
+	}
+	tx, err = r.WZeta.Approve(r.ZEVMAuth, r.ZevmTestDAppAddr, amount)
+	if err != nil {
+		panic(err)
+	}
+	r.Logger.Info("wzeta approve tx hash: %s", tx.Hash().Hex())
+	receipt = utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+	r.Logger.EVMReceipt(*receipt, "wzeta approve")
+	if receipt.Status == 0 {
+		panic(fmt.Sprintf("approve failed, logs: %+v", receipt.Logs))
+	}
+	testDAppZEVM, err := testdapp.NewTestDApp(r.ZevmTestDAppAddr, r.ZEVMClient)
```

Comment:
I may think we already deploy testDapp in local E2E initialization. Does it need to be redeployed for every tests?